src/Floor.java

```diff
@@ -1,4 +1,62 @@
-
+/**
+ * The floor class is responsible for tracking the floor number and also
+ * toggling the lights for the up/down buttons.
+ *
+ */
 public class Floor {
+	private int floorNumber;
+	private FloorButton upButton;
+	private FloorButton downButton;
+
+	/**
+	 * Constructor for Floor.
+	 * 
+	 * @param floorNumber - The current floor of the elevator.
+	 */
+	public Floor(int floorNumber) {
+		this.floorNumber = floorNumber;
+	}
+
+	/**
+	 * Turns on the Lamp depending on the direction of the elevator. If the elevator
+	 * is going up, then the lamp will turn on the up lamp.
+	 * 
+	 * @param direction
+	 * @param lampStatus
+	 */
+	public void turnButtonOn(Direction direction, boolean lampStatus) {
+		if (direction == Direction.UP) {
+			upButton.switchOn(lampStatus);
+		} else {
+			downButton.switchOn(lampStatus);
+		}
+	}
+
+	/**
+	 * Getter for the upButton lampStatus
+	 * 
+	 * @return true if lamp is on
+	 */
+	public boolean getLampStatusForUpButton() {
```

Comment:
same here, this method should be called something like "isUpButtonOn" instead of referring to the lampstatus. End user shouldn't know that buttons have lamps